frontend/tests/playwright/msw/request-handler.ts

```diff
@@ -86,148 +110,121 @@ export type MockedResponseInfo = {
-
-                  return res(ctx.status(status), ctx.data(body));
-               })
-         );
-      } else if (isRestMethod(method)) {
-         return new RestHandler(
-            method,
-            endpoint,
-            customResolver ??
-               ((req, res, ctx) => {
-                  if (passthrough) return req.passthrough();
-
-                  const transformers = [ctx.status(status)];
-
-                  if (headers) {
-                     transformers.push(ctx.set(headers));
-                  }
-
-                  // Only transform the body if it and the response type are defined
-                  if (body && responseType) {
-                     /**
-                      * `raw` is for our own naming convention, but the actual
-                      * transformer method is called `body`. This can be confusing
-                      * so we are using our own naming convention to make it
-                      * clearer what response type we will be using.
-                      * @see https://mswjs.io/docs/api/context/body
-                      */
-                     transformers.push(
-                        ctx[responseType === 'raw' ? 'body' : responseType](
-                           body
-                        )
-                     );
-                  }
-
-                  if (once) return res.once(...transformers);
-
-                  return res(...transformers);
-               })
-         );
-      } else {
-         throw new Error(
-            `Invalid method ${method}. Must be 'mutation', 'query', 'all', 'get', 'post', 'put', 'patch', or 'delete'.`
-         );
-      }
-   }
-}
+/**
+ * @property `customResolver` - If provided, this will be used as the resolver
+ * for the handler. This is useful if you want to do something more complex than
+ * just returning a static response.
+ * @see https://mswjs.io/docs/basics/response-resolver
+ */
+export type RestHandlerOptions =
+   | HandlerOptions
+   | { customResolver: RestResolver };
 
 /**
- * This is a type guard to ensure that the `method` is a valid GraphQLMethod
- * by using Typescript Predicate Return Types.
- * @see https://www.typescriptlang.org/docs/handbook/2/classes.html#this-based-type-guards
+ * Create a new GraphQL handler.
  */
-function isGraphQLMethod(method: string): method is GraphQLMethods {
-   return method === 'mutation' || method === 'query';
-}
+export const createGraphQLHandler = ({
+   request,
+   response,
+   options,
+}: {
+   request: GraphQLRequestInfo;
+   response: GraphQLResponseInfo;
+   options?: GraphQLHandlerOptions;
+}): GraphQLHandler => {
+   const { endpoint, method } = request;
+   const { status, body } = response;
+
+   const customResolver =
+      typeof options === 'function' ? (options as GraphQLResolver) : null;
+   const { once = false, passthrough = false } =
+      typeof options === 'object' ? (options as HandlerOptions) : {};
```

Comment:
Not a big deal, but have you tried leaving these casts out? I'd have expected TypeScript to potentially be able to infer the types here, and doing that would probably be safer for the long-term.

Reply:
Unfortunately, Typescript does not implicitly understand that `once` and `passthrough` exists in `options`

![image](https://user-images.githubusercontent.com/22064420/219523640-1b799eb6-152d-4cf9-87f7-d6fa11aeeeda.png)

And `customResovler` would resolve to `null` instead of either `RestResolver` or `null`

![image](https://user-images.githubusercontent.com/22064420/219523761-1b310aa9-2a80-4255-8f8b-b6d15d351d92.png)